### PR TITLE
video-as-img loses alpha channel

### PR DESCRIPTION
**Pull request by jernoble**
#### 0962e25391d210a563b4ec52bee8b8d6e6e32080
<pre>
video-as-img loses alpha channel
<a href="https://bugs.webkit.org/show_bug.cgi?id=243105">https://bugs.webkit.org/show_bug.cgi?id=243105</a>
&lt;rdar://96582168&gt;

Reviewed by Geoffrey Garen.

In 251557@main, a step was added where the results of a image decode were drawn to
a ShareableBitmap to facilitate transferring the results between the GPU and WebContent
processes. This bitmap was explicitly set to be &quot;opaque&quot;. However, the results of decoding
a frame of video will always generate a CGImage with valid alpha channel information, even
if that information is fully opaque, and in the case where the source video content has a
valid alpha channel, this information is lost at this drawing step.

* LayoutTests/media/content/video-with-alpha.mov: Added.
* LayoutTests/media/video-as-img-with-alpha-expected.html: Added.
* LayoutTests/media/video-as-img-with-alpha.html: Added.
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):

Canonical link: <a href="https://commits.webkit.org/252836@main">https://commits.webkit.org/252836@main</a>
</pre>

### CODE REVIEW

File: Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ void RemoteImageDecoderAVFProxy::createFrameImageAtIndex(ImageDecoderIdentifier
         return;
     DestinationColorSpace colorSpace { CGImageGetColorSpace(frameImage.get()) };
 
-    auto bitmap = ShareableBitmap::create(IntSize(width, height), { WTFMove(colorSpace), true /* isOpaque */ });
+    auto bitmap = ShareableBitmap::create(IntSize(width, height), { WTFMove(colorSpace), false /* isOpaque */ });
```
Comment (geoffreygaren) [inline Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp]: I prefer the style of making a local variable "bool isOpaque = false". I think it reads better at the call site.

**Review by geoffreygaren — APPROVED** [review]
r=me

**Comment by jernoble** [comment]
I'm going to be away from the office this afternoon. Please feel free to add the `merge-queue` label once all ~30 checks have passed.

**Comment by jernoble** [comment]
The `mac-wk1` test failure is super weird; the normal div spacer between the img and canvas elements is missing.

**Comment by jernoble** [comment]
> The `mac-wk1` test failure is super weird; the normal div spacer between the img and canvas elements is missing.

Aha, it happened because of the order of two `<script>` statements. In WK2, the inline `<script>` ran first, regardless of the order; in WK1, the inline `<script>` ran second.  Updated the test and will add the merge keyword (since the only change is in the test itself).

**Pull request merged**